anti_patterns/deny-warnings.md

```diff
@@ -0,0 +1,101 @@
+# `#![deny(warnings)]`
+
+## Description
+
+A well-intentioned crate author wants to ensure their code builds without
+warnings. So they annotate their crate root with
+
+## Example
+
+```rust
+#![deny(warnings)]
+
+// All is well.
+```
+
+## Advantages
+
+It is short and will stop the build if anything is amiss.
+
+## Drawbacks
+
+By disallowing the compiler to build with warnings, a crate author opts out of
+Rust's famed stability. Sometimes new features or old misfeatures need a change
```

Comment:
I would like an example of such a change -- any lint from the `future-incompatible` group would be okay.